src/core/lib/gprpp/status_helper.h

```diff
@@ -0,0 +1,147 @@
+  STREAM_ID,
+  /// grpc status code representing this error
+  // TODO(veblush): Remove this after grpc_error is replaced with absl::Status
+  GRPC_STATUS,
+  /// offset into some binary blob (usually represented by
+  /// RAW_BYTES) where the error occurred
+  OFFSET,
+  /// context sensitive index associated with the error
+  INDEX,
+  /// context sensitive size associated with the error
+  SIZE,
+  /// http2 error code associated with the error (see the HTTP2 RFC)
+  HTTP2_ERROR,
+  /// TSI status code associated with the error
+  TSI_CODE,
+  /// WSAGetLastError() reported when this error occurred
+  WSA_ERROR,
+  /// File descriptor associated with this error
+  FD,
+  /// HTTP status (i.e. 404)
+  HTTP_STATUS,
+  /// chttp2: did the error occur while a write was in progress
+  OCCURRED_DURING_WRITE,
+  /// channel connectivity state associated with the error
+  CHANNEL_CONNECTIVITY_STATE,
+  /// LB policy drop
+  LB_POLICY_DROP,
+};
+
+/// This enum should have the same value of grpc_error_strs
+// TODO(veblush): Use camel-case names once migration to absl::Status is done.
+enum class StatusStrProperty {
+  /// top-level textual description of this error
+  DESCRIPTION,
+  /// source file in which this error occurred
+  FILE,
+  /// operating system description of this error
+  OS_ERROR,
+  /// syscall that generated this error
+  SYSCALL,
+  /// peer that we were trying to communicate when this error occurred
+  TARGET_ADDRESS,
+  /// grpc status message associated with this error
+  GRPC_MESSAGE,
+  /// hex dump (or similar) with the data that generated this error
+  RAW_BYTES,
+  /// tsi error string associated with this error
+  TSI_ERROR,
+  /// filename that we were trying to read/write when this error occurred
+  FILENAME,
+  /// key associated with the error
+  KEY,
+  /// value associated with the error
+  VALUE,
+  /// time string to create the error
+  CREATED_TIME,
+};
+
+/// Creates a status with given additional information
+absl::Status StatusCreate(
+    absl::StatusCode code, absl::string_view msg, const DebugLocation& location,
+    std::initializer_list<absl::Status> children) GRPC_MUST_USE_RESULT;
+
+/// Sets the int property to the status
+void StatusSetInt(absl::Status* status, StatusIntProperty key, intptr_t value);
+
+/// Gets the int property from the status
+absl::optional<intptr_t> StatusGetInt(
+    const absl::Status& status, StatusIntProperty key) GRPC_MUST_USE_RESULT;
+
+/// Sets the str property to the status
+void StatusSetStr(absl::Status* status, StatusStrProperty key,
+                  absl::string_view value);
+
+/// Gets the str property from the status
+absl::optional<std::string> StatusGetStr(
+    const absl::Status& status, StatusStrProperty key) GRPC_MUST_USE_RESULT;
+
+/// Adds a child status to status
+void StatusAddChild(absl::Status* status, absl::Status child);
+
+/// Returns all children status from a status
+std::vector<absl::Status> StatusGetChildren(absl::Status status)
```

Comment:
I'm trying to [upgrade google-cloud-cpp to grpc v1.38.0](https://github.com/googleapis/google-cloud-cpp/pull/6621) and the build breaks because this file doesn't include `<vector>`

The errors from [our build log](https://storage.googleapis.com/cloud-cpp-community-publiclogs/logs/google-cloud-cpp/6621/ae762e2afbff8375f2cb74af2908bdfd3f257ae9/fedora-33-bazel-targets/log-0d856e2f-62be-429e-86b2-9bc4e21a8ae5.txt):

```
Step #3: In file included from external/com_github_grpc_grpc/src/core/lib/iomgr/error.h:33,
Step #3:                  from external/com_github_grpc_grpc/src/core/lib/iomgr/closure.h:33,
Step #3:                  from external/com_github_grpc_grpc/src/core/lib/channel/handshaker.h:33,
Step #3:                  from external/com_github_grpc_grpc/src/core/lib/security/security_connector/security_connector.h:28,
Step #3:                  from external/com_github_grpc_grpc/src/core/lib/security/security_connector/ssl/ssl_security_connector.h:26,
Step #3:                  from external/com_github_grpc_grpc/src/core/lib/security/security_connector/ssl/ssl_security_connector.cc:21:
Step #3: external/com_github_grpc_grpc/src/core/lib/gprpp/status_helper.h:139:6: error: 'vector' in namespace 'std' does not name a template type
Step #3:   139 | std::vector<absl::Status> StatusGetChildren(absl::Status status)
Step #3:       |      ^~~~~~
Step #3: external/com_github_grpc_grpc/src/core/lib/gprpp/status_helper.h:25:1: note: 'std::vector' is defined in header '<vector>'; did you forget to '#include <vector>'?
Step #3:    24 | #include "absl/status/status.h"
Step #3:   +++ |+#include <vector>
Step #3:    25 | #include "absl/time/time.h"
Step #3: ERROR: /h/.cache/bazel/_bazel_root/9bd401995a189c62294f86ba79716c4c/external/com_github_grpc_grpc/BUILD:223
```